### PR TITLE
[LP 412] add link errors to landing page

### PR DESCRIPTION
**Pull request by lindawoo-plaid**
Jira ticket:  https://jira.plaid.com/browse/LP-412

This adds information on link token create errors to landing page.  The first message will still appear if the server is not running.  Otherwise if an error occurs in link_create_token, error messages will be displayed.

before: 
image

after: 

image

### CODE REVIEW

**Review by phoenixy1 — APPROVED** [review]
Approving from a behavior POV; did not review for code quality

File: frontend/src/Components/Headers/index.tsx [hunk 1]
```diff
@@ -35,12 +38,28 @@ const Header = () => {
...
           {/* message if backend is not running and there is no link token */}
-          {linkToken == null || backend === false ? (
+          {backend === false ? (
             <Callout warning>
               Unable to fetch link_token: please make sure your backend server
               is running and that your .env file has been configured with your
               <code>PLAID_CLIENT_ID</code> and <code>PLAID_SECRET</code>.
             </Callout>
+          ) : /* message if backend is running and there is no link token */
+          linkToken == null && backend === true ? (
```
Comment (skylarmb) [inline frontend/src/Components/Headers/index.tsx:48]:
```suggestion
          linkToken == null && backend ? (
```

File: frontend/src/Components/Headers/index.tsx [hunk 2]
```diff
@@ -35,12 +38,28 @@ const Header = () => {
             to access their accounts via the Plaid API.
           </p>
           {/* message if backend is not running and there is no link token */}
-          {linkToken == null || backend === false ? (
+          {backend === false ? (
```
Comment (skylarmb) [inline frontend/src/Components/Headers/index.tsx:41]:
```suggestion
          {!backend ? (
```

File: frontend/src/App.tsx [hunk 3]
```diff
@@ -43,6 +43,18 @@ const App = () => {
...
       const data = await response.json();
       if (data) {
+        if (data.error != null) {
+          dispatch({
+            type: "SET_STATE",
+            state: {
+              linkToken: null,
+              linkErrorMessage: data.error.error_message,
+              linkErrorType: data.error.error_type,
+              linkErrorCode: data.error.error_code,
```
Comment (skylarmb) [inline frontend/src/App.tsx:53]: These variable names are a bit confusing to me as this error is not a Link error (which is something else), it's a Link token API error right?. Maybe we could just add one key to the state that represents the actual error object and do something like this

```suggestion
              linkTokenError: data.error,
```

then later in the template 

```jsx

Error Code: <code>{linkTokenError.error_code}</code>

```

etc

**Review by skylarmb — CHANGES_REQUESTED** [review]

**Review by skylarmb — APPROVED** [review]

**Pull request merged**